.github/workflows/deploy.yml

```diff
@@ -27,17 +27,31 @@ jobs:
   deploy:
     name: Deploy
     runs-on: ubuntu-latest
+    container:
+      image: ubuntu:22.04 # Ubuntu 22.04 is the latest LTS version as of April 21, 2022: https://wiki.ubuntu.com/Releases
```

Comment:
Can we try reverting this and changes related to this? (e.g.: explicit `apt-get`, `ImageOS: ubuntu22` and so on)

If we can use the `ubuntu-latest` image directly, we can use simpler `deploy.yml` and don't maintain Ubuntu version for this.

Reply:
I'm OK with just using `ubuntu-latest` and no container. However, I know there were some concerns about this approach expressed by @avantgardnerio (e.g. the proprietary nature and lack of debuggability of `ubuntu-latest`.

However, I can try this approach and if no one else in the community has reservations about it, then I am OK with it.

Reply:
Ah, I can understand the debuggability concern. If we make this job local debuggable, we should not use `actions/*` in `ubuntu:XXX` container like we did in apache/arrow: https://github.com/apache/arrow/blob/main/.github/workflows/cpp.yml#L92-L99

If we need local debuggability, we may need to choose another approach.

Reply:
Thanks.
It seems to work. I prefer this approach to the no container approach but I want to hear opinions from others.

Reply:
This worked fine: https://github.com/mathworks/arrow-site/actions/runs/4321911014/jobs/7543691983.

Should we use this approach instead despite the potential tradeoffs?

**Note**: I'm not sure if the the `npm` package caching is working as expected. I see https://github.com/mathworks/arrow-site/actions/runs/4321911014/jobs/7543691983. I also see https://github.com/mathworks/arrow-site/actions/runs/4321911014/jobs/7543691983#step:8:32 which is only taking 2 seconds. So maybe it is working.

Reply:
Sorry, it seems like I was writing a comment at the same time you were.

Thanks - I'll wait to hear from others before making any further changes.

Reply:
I don't fully understand all the tradeoffs involved, but I do think it is important to be able to build the site locally without having to install the dependencies directly -- here are instructions for using docker to do so https://github.com/apache/arrow-site#using-docker

Reply:
Sadly I don't love any of the options here:

- Use ubuntu-latest (which has node 18) - not reproducible locally
- Use docker:22.04, but use the github action to install a more recent node - not reproducible locally
- Add a deb package source and install via apt - fine in docker, not recommended locally
- Use snap to install node - wonderful locally, but doesn't work in docker
- Install node version manager (nvm) in docker - maybe the best option?
- curl node and install the tarballs from the binary distribution

Sadly, this is one of those cases where what I would recommend in docker (probably [this](https://github.com/nodesource/distributions/blob/master/README.md#debinstall)) doesn't match what I would do locally (sudo snap install nodejs).

Reply:
It seems that we use "locally" as different meanings.
It seems that @avantgardnerio uses "locally" as "on my machine without Docker" and others (including me) use "locally" as "on my machine with Docker".

@avantgardnerio Is "without Docker" important for you? Is "with Docker" acceptable?

Reply:
Oh, sorry - to clarify, yes I run Ubuntu 22:04 natively on my laptop.

Yes, I'd expect to be able to continue to run natively on my laptop, but no I don't mind running pasting in some commands from the CI script to set my machine up to do it.

Reply:
Thanks for confirming this.

How about the following steps to improve the current situation?

1. We merge "no container" version https://github.com/mathworks/arrow-site/tree/GH-325-no-container because it's simpler than this version
2. We create a `Dockerfile` based on `ubuntu:22.04` and use it in CI and https://github.com/apache/arrow-site#using-docker (This prevents breaking @alamb 's use case accidentally)
3. We extract command lines used in 2. as a shell script and fine tune the shell script to work on host Ubuntu 22.04 too (@avantgardnerio 's use case)

Reply:
Thanks @avantgardnerio and @kou for your sharing your thoughts! I agree. This sounds like a reasonable path forward.

Just to clarify:

"We create a Dockerfile based on ubuntu:22.04 and **use it in CI**"

By "**use it in CI**" do you mean that you would like the GitHub Actions workflow to run inside of a Docker container based on this `Dockerfile`? If so, doesn't this conflict with 1. (which is to use the "no container" workflow)?

Sorry for the confusion. I just want to make sure we are all on the same page before moving forward.

Reply:
My idea is that we just use `docker build && docker run` in GitHub Actions instead of `container:`.

(More specifically, we will use `docker/build-push-action` instead of raw `docker build` and push a built image to `ghcr.io` to reuse the built image on local for https://github.com/apache/arrow-site#using-docker . https://github.com/groonga/groonga-delta/blob/main/.github/workflows/docker.yml is a bit old but may help you to understand my idea.)

Reply:
Thanks for clarifying @kou! This make sense!

I'll start working on these changes now.